lib/go/edgecontext/edgecontext_test.go

```diff
@@ -769,6 +784,81 @@ func TestFromHeader(t *testing.T) {
 			}
 		},
 	)
+
+	t.Run(
+		"service",
+		func(t *testing.T) {
+			t.Run("service auth only", func(t *testing.T) {
+				e, err := edgecontext.FromHeader(context.Background(), headerWithValidServiceAuth, globalTestImpl)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				svc, ok := e.Service()
+				if !ok {
+					t.Errorf("Expected service to be true, got false")
+				}
+				if name, ok := svc.Name(); !ok {
+					t.Error("Failed to get service name")
```

Comment:
you can use t.Fatalf and then you don't need the else, it can just be another if